packages/cache-redis/src/services/redis-cache.ts

```diff
@@ -0,0 +1,84 @@
+import { Redis } from "ioredis"
+import {
+  ConfigurableModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
+  ICacheService,
+} from "@medusajs/medusa"
+import { MedusaError } from "medusa-core-utils"
+import { RedisCacheModuleOptions } from "../types"
+
+const DEFAULT_CACHE_TIME = 30 // 30 seconds
+const EXPIRY_MODE = "EX" // "EX" stands for an expiry time in second
+
+type InjectedDependencies = {
+  redisClient: Redis
+}
+
+class RedisCacheService implements ICacheService {
+  protected TTL: number
+  protected readonly redis: Redis
+
+  constructor(
+    { redisClient }: InjectedDependencies,
+    options: RedisCacheModuleOptions = {},
+    moduleDeclaration?: ConfigurableModuleDeclaration
+  ) {
+    this.redis = redisClient
+    this.TTL = options.defaultTTL || DEFAULT_CACHE_TIME
+
+    if (moduleDeclaration?.resources !== MODULE_RESOURCE_TYPE.SHARED) {
```

Comment:
I think we can also support "isolated" resources here.
I can apply the changes after the PR is merged if you prefer.
[Here](https://github.com/medusajs/medusa/pull/3329/files#diff-d2d3c280dba3a849f2a36c762afcef78f9a556cf3c94b3d261b6425898bb5ac8) is an example of the loader creating and registering it's own connection.

Reply:
In a somewhat similar context, should we remove the dependency on the `redisClient` from the core and use the same approach as in the Redis event bus i.e. create a dedicated connection in the module loader?

You can check it out in [this commit](https://github.com/medusajs/medusa/pull/2599/commits/1753cc20c266eacb2926079281c7b22d14b5b50a). 

@fPolic what do you think?

Reply:
yes, this makes more sense 👍 
I've added the loader in https://github.com/medusajs/medusa/pull/3187/commits/2a092e5b407687f41487901bf5b33b90b69a8861